### PR TITLE
fix CSF: strip trailing spaces/tabs during normalization for hashing

### PR DESCRIPTION
**Pull request by hko-s**
Added a test to debug #424

### CODE REVIEW

**Comment by wiktor-k** [comment]
Hmm... maybe it'd be good to also attach a test case *without* the space to be trimmed? :thinking: 

Looks good to me. You could add "Fixes #424" in the commit message.

**Comment by hko-s** [comment]
> Hmm... maybe it'd be good to also attach a test case _without_ the space to be trimmed? 🤔

We already have other CSF test messages that get tested. I think/hope those cover the base cases.
However, this case was totally missing.

Now that you mention it, though, I could add another test that does a sign/verify roundtrip on this message (the current test only does the "verify" side).

> Looks good to me. You could add "Fixes #424" in the commit message.

Right. Thanks, done!

File: src/composed/cleartext.rs [hunk 1]
```diff
@@ -305,18 +305,35 @@ fn dash_escape(text: &str) -> String {
     out
 }
 
-/// Undo dash escaping of `text`.
+/// Undo dash escaping of `text`, and trim space/tabs at the end of lines.
 ///
-/// This implementation is implicitly agnostic between "\n" and "\r\n" line endings.
-fn dash_unescape(text: &str) -> String {
+/// This implementation can handle both "\n" and "\r\n" line endings.
+fn dash_unescape_and_trim(text: &str) -> String {
```
Comment (dignifiedquire) [inline src/composed/cleartext.rs:311]: it would be good to have unit test for this behaviour, just testing this function

Comment (hko-s) [reply src/composed/cleartext.rs:311]: good point. i've added a unit test for just the unescaper.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**